src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile_isolated_strategy.py

```diff
@@ -38,6 +40,15 @@ def __init__(self, context, options, workdir, analysis_tools, language, sources_
     self._analysis_dir = os.path.join(workdir, 'isolated-analysis')
     self._classes_dir = os.path.join(workdir, 'isolated-classes')
 
+    # TODO: Regester delete_scratch in JvmCompile
+    self._delete_scratch = options.delete_scratch
```

Comment:
You aren't making use of this value.  I think you should delete the scratch files when everything is finished (unless this value is false)